payu/models/fms.py

```diff
@@ -125,28 +134,36 @@ def collate(self):
             collate_ignore = [collate_ignore]
 
         # Generate collated file list and identify the first tile
-        tile_fnames = [f for f in os.listdir(self.output_path)
-                       if f[-4:].isdigit() and f[-8:-4] == '.nc.']
+        tile_fnames = {}
+        tile_fnames[self.output_path] = Fms.get_uncollated_files(self.output_path)
 
-        tile_fnames.sort()
+        print(tile_fnames)
+
+        if collate_config.get('restart',False):
+            # Add uncollated restart files 
+            tile_fnames[self.prior_restart_path] = Fms.get_uncollated_files(self.prior_restart_path)
 
-        mnc_tiles = defaultdict(list)
-        for t_fname in tile_fnames:
-            t_base, t_ext = os.path.splitext(t_fname)
-            t_ext = t_ext.lstrip('.')
+        # mnc_tiles = defaultdict(list)
+        mnc_tiles = defaultdict(defaultdict(list).copy)
+        for t_dir in tile_fnames:
+            for t_fname in tile_fnames[t_dir]:
+                t_base, t_ext = os.path.splitext(t_fname)
+                t_ext = t_ext.lstrip('.')
 
-            # Skip any files listed in the ignore list
-            if t_base in collate_ignore:
-                continue
+                # Skip any files listed in the ignore list
+                if t_base in collate_ignore:
+                    continue
 
-            mnc_tiles[t_base].append(t_fname)
+                mnc_tiles[t_dir][t_base].append(t_fname)
+
+        # print(mnc_tiles)
 
         if mpi and collate_config.get('glob', True):
             for t_base in mnc_tiles:
                 globstr = "{}.*".format(t_base)
                 # Try an equivalent glob and check the same files are returned
-                mnc_glob = fnmatch.filter(os.listdir(self.output_path,
-                                                     globstr))
+                mnc_glob = fnmatch.filter(os.listdir(self.output_path),
```

Comment:
This was the error I was referring to @marshallward

Reply:
oops sorry about that

Reply:
No worries. I didn't test it obviously.